procrastinate/testing.py

```diff
@@ -296,6 +296,22 @@ async def select_stalled_jobs_all(self, nb_seconds, queue, task_name):
             and task_name in (job["task_name"], None)
         )
 
+    async def select_stalled_jobs_by_heartbeat_all(
+        self, queue, task_name, seconds_since_heartbeat
+    ):
+        return (
+            job
+            for job in self.jobs.values()
+            if job["status"] == "doing"
+            and queue in (job["queue_name"], None)
+            and task_name in (job["task_name"], None)
+            and (
+                self.heartbeats.get(job["worker_id"]) is None
+                or self.heartbeats[job["worker_id"]]
+                < utils.utcnow() - datetime.timedelta(seconds=seconds_since_heartbeat)
+            )
```

Comment:
We could do 
```
self.heartbeats.get(job["worker_id"], datetime.min)
```
and avoid having to handle the None case (probably `.min.replace(tzinfo=utc)` though if utcnow is tz aware))